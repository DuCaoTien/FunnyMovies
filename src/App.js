import React from 'react'
import { Routes, Route } from 'react-router-dom';
import { ToastContainer } from "react-toastify";

import Dashboard from "./pages/content/dashboard";
import Header from "./pages/content/header";
import SharedMovie from "./pages/content/shared-movie";
import { ProtectedRoute } from "./routes/ProtectedRoute";

import "./scss/main.scss";

function App() {
    return (
        <div className="App">
            <React.Fragment>
                <Header/>
                <Routes>
                    <Route exact path='*' element={<Dashboard/>}/>
                    <Route exact path='/FunnyMovies/' element={<Dashboard/>}/>
                    <Route
                        exact
                        path='/FunnyMovies/share'
                        element={
                            <ProtectedRoute>
                                <SharedMovie/>
                            </ProtectedRoute>
                        }
                    />
                </Routes>
                <ToastContainer position="top-left" />
            </React.Fragment>
        </div>
    );
}

export default App;
