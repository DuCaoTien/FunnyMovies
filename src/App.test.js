import React from 'react';
import {render, act} from "@testing-library/react";

import App from './App';
import {MemoryRouter} from "react-router-dom";

const mockedNavigator = jest.fn();
const mockedRoutes = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedNavigator,
    useRoutes: () => mockedRoutes,
    useLocation: jest.fn()
}));


test('App runs as expected', async () => {
    const badRoute = '/FunnyMovies/'

    let rendered = null;

    await act(async () => {
        rendered = render(
            <MemoryRouter initialEntries={[badRoute]}>
                <App/>
            </MemoryRouter>,
        )
    })

    expect(rendered).toMatchSnapshot();
})
