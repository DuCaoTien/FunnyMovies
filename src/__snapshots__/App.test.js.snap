// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App runs as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <div
            class="header_content"
          >
            <div
              class="header_side"
            >
              <div
                class="logo"
              >
                FUNNY MOVIES
              </div>
            </div>
            <div
              class="header_side"
            >
              <div
                class="signin"
              >
                <div>
                  <input
                    autocomplete="email"
                    class="input"
                    id="email"
                    name="email"
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
                <div>
                  <input
                    autocomplete="new-password"
                    class="input"
                    id="password"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  class="submit"
                >
                  <button
                    aria-label="login"
                    class="btn"
                    disabled=""
                    name="btn_login"
                    tabindex="0"
                  >
                    Login
                  </button>
                </div>
                <div
                  class="submit"
                >
                  <button
                    aria-label="register"
                    class="btn"
                    disabled=""
                    name="btn-register"
                    tabindex="0"
                  >
                    Register
                  </button>
                </div>
              </div>
            </div>
          </div>
        </header>
        <div
          class="content"
        >
          <div
            class="row"
          >
            <div
              class="side"
            >
              <div
                class="bg"
              />
              <iframe
                allow="accelerometer; autoplay;encrypted-media; gyroscope;picture-in-picture"
                allowfullscreen=""
                frameborder="0"
                height="340px"
                loading="lazy"
                src="https://www.youtube.com/embed/ql3QS4bzcWg"
                title="https://www.youtube.com/embed/ql3QS4bzcWg"
                width="480px"
              />
            </div>
            <div
              class="side"
            >
              <div
                class="w100 info"
              >
                <div
                  class="title"
                >
                  Đen - Ta Cứ Đi Cùng Nhau ft. Linh Cáo (Prod. by I TEU) [Official Audio]
                </div>
                <div
                  class="subTitle"
                >
                  <span>
                    Shared by: 
                  </span>
                  <span>
                    Đen Vâu Official
                  </span>
                </div>
              </div>
              <div
                class="w100 action"
              >
                <button
                  aria-label="voted-up"
                  class="btn "
                  name="voted-up"
                  type="submit"
                >
                  <span>
                    27746
                  </span>
                  <img
                    alt="like"
                    src="like.png"
                  />
                </button>
              </div>
              <div
                class="w100 des"
              >
                <div
                  class="label"
                >
                  Description: 
                </div>
                <p>
                  Ta Cứ Đi Cùng Nhau - Đen ft. Linh Cáo
Download/Stream: https://Denvau.lnk.to/TCDCN

Prod. by Nam Phong Truong & D.A From #iTeu
Recording: CWave Studio & SOS Studio
Mix & Mastering by Trác Ngọc Lĩnh @ ACK Studio
Credit to http://www.freesfx.co.uk for the Ocean Waves, Children and Seagull sounds.

follow Linh Cáo:
/youtube: https://www.youtube.com/channel/UCHkZEgtTon7Dliw2NYPyjRQ
/facebook:  https://www.facebook.com/thuylinh.nguyen.7161
/instagram: https://www.instagram.com/linhcao_021190/

follow Đen:
/instagram: https://www.instagram.com/den.vau/
/youtube :    http://www.youtube.com/c/DenVau1305
/spotify:       https://spoti.fi/2BAHtnX
/itunes:        https://apple.co/2zCT9oQ
/facebook:
//////page:    https://www.facebook.com/denvau
////profile:    https://www.facebook.com/denvau1305
#tacudicungnhau #den #linhcao

© 2017 Đen
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Toastify"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <div
          class="header_content"
        >
          <div
            class="header_side"
          >
            <div
              class="logo"
            >
              FUNNY MOVIES
            </div>
          </div>
          <div
            class="header_side"
          >
            <div
              class="signin"
            >
              <div>
                <input
                  autocomplete="email"
                  class="input"
                  id="email"
                  name="email"
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
              <div>
                <input
                  autocomplete="new-password"
                  class="input"
                  id="password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="submit"
              >
                <button
                  aria-label="login"
                  class="btn"
                  disabled=""
                  name="btn_login"
                  tabindex="0"
                >
                  Login
                </button>
              </div>
              <div
                class="submit"
              >
                <button
                  aria-label="register"
                  class="btn"
                  disabled=""
                  name="btn-register"
                  tabindex="0"
                >
                  Register
                </button>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="content"
      >
        <div
          class="row"
        >
          <div
            class="side"
          >
            <div
              class="bg"
            />
            <iframe
              allow="accelerometer; autoplay;encrypted-media; gyroscope;picture-in-picture"
              allowfullscreen=""
              frameborder="0"
              height="340px"
              loading="lazy"
              src="https://www.youtube.com/embed/ql3QS4bzcWg"
              title="https://www.youtube.com/embed/ql3QS4bzcWg"
              width="480px"
            />
          </div>
          <div
            class="side"
          >
            <div
              class="w100 info"
            >
              <div
                class="title"
              >
                Đen - Ta Cứ Đi Cùng Nhau ft. Linh Cáo (Prod. by I TEU) [Official Audio]
              </div>
              <div
                class="subTitle"
              >
                <span>
                  Shared by: 
                </span>
                <span>
                  Đen Vâu Official
                </span>
              </div>
            </div>
            <div
              class="w100 action"
            >
              <button
                aria-label="voted-up"
                class="btn "
                name="voted-up"
                type="submit"
              >
                <span>
                  27746
                </span>
                <img
                  alt="like"
                  src="like.png"
                />
              </button>
            </div>
            <div
              class="w100 des"
            >
              <div
                class="label"
              >
                Description: 
              </div>
              <p>
                Ta Cứ Đi Cùng Nhau - Đen ft. Linh Cáo
Download/Stream: https://Denvau.lnk.to/TCDCN

Prod. by Nam Phong Truong & D.A From #iTeu
Recording: CWave Studio & SOS Studio
Mix & Mastering by Trác Ngọc Lĩnh @ ACK Studio
Credit to http://www.freesfx.co.uk for the Ocean Waves, Children and Seagull sounds.

follow Linh Cáo:
/youtube: https://www.youtube.com/channel/UCHkZEgtTon7Dliw2NYPyjRQ
/facebook:  https://www.facebook.com/thuylinh.nguyen.7161
/instagram: https://www.instagram.com/linhcao_021190/

follow Đen:
/instagram: https://www.instagram.com/den.vau/
/youtube :    http://www.youtube.com/c/DenVau1305
/spotify:       https://spoti.fi/2BAHtnX
/itunes:        https://apple.co/2zCT9oQ
/facebook:
//////page:    https://www.facebook.com/denvau
////profile:    https://www.facebook.com/denvau1305
#tacudicungnhau #den #linhcao

© 2017 Đen
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Toastify"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
